File: Sources/LectureCopilot/CopilotAction.swift
import Foundation

enum CopilotAction {
    case translate
    case explain
    case directAnswer
    case sayInClass
    case backToClass
    case classSummary

    var promptKey: String? {
        switch self {
        case .translate: "translate"
        case .explain: "explain"
        case .directAnswer: "directAnswer"
        case .sayInClass: "sayInClass"
        case .classSummary: "classSummary"
        case .backToClass: nil
        }
    }

    var displayName: String {
        switch self {
        case .translate: "Translate"
        case .explain: "Explain"
        case .directAnswer: "Direct Answer"
        case .sayInClass: "Say in Class"
        case .backToClass: "Back to Class"
        case .classSummary: "Class Summary"
        }
    }

    var chipTitle: String {
        switch self {
        case .translate: "Translate"
        case .explain: "Explain"
        case .directAnswer: "Answer"
        case .sayInClass: "Speak"
        case .backToClass: "Class"
        case .classSummary: "Summary"
        }
    }
}

enum HotKeyEvent {
    case shiftLeft
    case shiftRight
    case shiftUp
    case shiftDown
    case returnKey
}

File: windows/LectureCopilot.Windows/Models/CopilotAction.cs
namespace LectureCopilot.Windows.Models;

public enum CopilotAction
{
    Translate,
    Explain,
    DirectAnswer,
    SayInClass,
    BackToClass,
    ClassSummary
}

public static class CopilotActionExtensions
{
    public static string? PromptKey(this CopilotAction action) => action switch
    {
        CopilotAction.Translate => "translate",
        CopilotAction.Explain => "explain",
        CopilotAction.DirectAnswer => "directAnswer",
        CopilotAction.SayInClass => "sayInClass",
        CopilotAction.ClassSummary => "classSummary",
        _ => null
    };

    public static string DisplayName(this CopilotAction action) => action switch
    {
        CopilotAction.Translate => "Translate",
        CopilotAction.Explain => "Explain",
        CopilotAction.DirectAnswer => "Direct Answer",
        CopilotAction.SayInClass => "Say in Class",
        CopilotAction.BackToClass => "Back to Class",
        CopilotAction.ClassSummary => "Class Summary",
        _ => action.ToString()
    };

    public static string ChipTitle(this CopilotAction action) => action switch
    {
        CopilotAction.DirectAnswer => "Answer",
        CopilotAction.SayInClass => "Speak",
        CopilotAction.BackToClass => "Class",
        CopilotAction.ClassSummary => "Summary",
        _ => action.DisplayName()
    };
}

public enum HotKeyEvent
{
    ShiftLeft,
    ShiftRight,
    ShiftUp,
    ShiftDown,
    ShiftReturn,
    ToggleHud
}
